packages/fxa-content-server/app/scripts/lib/payment-server.js

```diff
@@ -4,38 +4,54 @@
 
 'use strict';
 
+import Url from './url';
+
 const PaymentServer = {
+  /**
+   * Navigate to the payments server after flushing metrics.
+   *
+   * @param {Object} view
+   * @param {Object} subscriptionsConfig
+   * @param {String} redirectPath
+   * @param {Object} [queryParams={}]
+   */
   navigateToPaymentServer(
     view,
     subscriptionsConfig,
     redirectPath,
-    queryParams
+    queryParams = {}
   ) {
     const {
       managementClientId,
       managementScopes,
       managementTokenTTL,
       managementUrl,
     } = subscriptionsConfig;
-    const queryString =
-      typeof queryParams === 'object' &&
-      Object.keys(queryParams)
-        .filter(k => queryParams[k] !== null && queryParams[k] !== '')
-        .map(
-          k => `${encodeURIComponent(k)}=${encodeURIComponent(queryParams[k])}`
-        )
-        .join('&');
-    const qs = queryString.length ? `?${queryString}` : '';
+
+    const { flowBeginTime, flowId } = view.metrics.getFlowEventMetadata();
+
```

Comment:
FWIW I see the in teh logs that `device_id` is used for an amplitude event, but not the flowBeginTime.

Reply:
I'll add that.